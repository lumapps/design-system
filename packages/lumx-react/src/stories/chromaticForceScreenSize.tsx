/* eslint-disable */
import React from 'react';
import isChromatic from 'chromatic/isChromatic';

/** Story decorator used to force a minimum screen size when running in chromatic */
export const chromaticForceScreenSize = (story: any) =>
    isChromatic() ? <div style={{ minWidth: 1200, minHeight: 800 }}>{story()}</div> : story();
