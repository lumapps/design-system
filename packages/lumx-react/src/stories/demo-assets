../../../site-demo/static/demo-assets