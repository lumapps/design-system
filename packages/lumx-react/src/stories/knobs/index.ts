export * from './image';
export * from './lorem';
