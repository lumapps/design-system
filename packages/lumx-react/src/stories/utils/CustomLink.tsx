/* eslint-disable jsx-a11y/anchor-has-content,jsx-a11y/anchor-is-valid */
import React from 'react';

/**
 * Example custom link that can be used in linkAs props.
 */
export const CustomLink: any = ({ ...props }) => <a href="#" data-custom-link {...props} />;
