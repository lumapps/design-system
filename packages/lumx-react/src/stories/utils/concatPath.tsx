export type PathPart = string | number;
export type Path = PathPart | Array<PathPart>;

/**
 * Concat flatten object path
 *
 * @example concatPath('foo', 'bar') // => 'foo.bar'
 * @example concatPath(['foo', 0]) // => 'foo[0]'
 * @example concatPath('foo', 0, ['bar']) // => 'foo[0].bar'
 */
export const concatPath = (...prefix: Path[]) => {
    const [first, ...rest] = prefix.flat();
    return rest.reduce<string>((acc, part) => {
        if (typeof part === 'number') return `${acc}[${part}]`;
        return `${acc}.${part}`;
    }, String(first));
};
