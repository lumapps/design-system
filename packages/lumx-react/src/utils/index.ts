/**
 * WARNING: All modules exported here are exposed to NPM in '@lumx/react/utils'.
 */

export { ClickAwayProvider } from './ClickAwayProvider';
export { Portal, type PortalProps, type PortalInit, PortalProvider, type PortalProviderProps } from './Portal';
