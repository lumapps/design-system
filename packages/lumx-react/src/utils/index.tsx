export * from '@lumx/core/js/utils';

export * from './getRootClassName';

export * from './htmlDecode';

export * from './partitionMulti';

export * from './type';
