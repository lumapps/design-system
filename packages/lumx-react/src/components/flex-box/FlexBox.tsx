import { Alignment, HorizontalAlignment, Orientation, VerticalAlignment } from '@lumx/react';
import { Comp, GenericProps, getRootClassName, handleBasicClasses } from '@lumx/react/utils';
import classNames from 'classnames';
import castArray from 'lodash/castArray';
import React, { forwardRef, ReactNode } from 'react';
import { Size } from '..';

export type MarginAutoAlignment = Extract<Alignment, 'top' | 'bottom' | 'right' | 'left'>;
export type GapSize = Extract<Size, 'regular' | 'big' | 'huge'>;

/**
 * Defines the props of the component.
 */
export interface FlexBoxProps extends GenericProps {
    /** Children elements. */
    children?: ReactNode;
    /** Whether the "content filling space" is enabled or not. */
    fillSpace?: boolean;
    /** Gap space between flexbox items. */
    gap?: GapSize;
    /** Flex horizontal alignment. */
    hAlign?: VerticalAlignment;
    /** Whether the "auto margin" is enabled all around or not. */
    marginAuto?: MarginAutoAlignment | MarginAutoAlignment[];
    /** Whether the "content shrink" is disabled or not. */
    noShrink?: boolean;
    /** Flex direction. */
    orientation?: Orientation;
    /** Flex vertical alignment. */
    vAlign?: HorizontalAlignment;
    /** Whether the "flex wrap" is enabled or not. */
    wrap?: boolean;
}

/**
 * Component display name.
 */
const COMPONENT_NAME = 'FlexBox';

/**
 * Component default class name and class prefix.
 */
const CLASSNAME = getRootClassName(COMPONENT_NAME);

/**
 * FlexBox component.
 *
 * @param  props Component props.
 * @param  ref   Component ref.
 * @return React element.
 */
export const FlexBox: Comp<FlexBoxProps, HTMLDivElement> = forwardRef((props, ref) => {
    const {
        children,
        className,
        fillSpace,
        gap,
        hAlign,
        marginAuto,
        noShrink,
        orientation,
        vAlign,
        wrap,
        ...forwardedProps
    } = props;

    return (
        <div
            ref={ref}
            {...forwardedProps}
            className={classNames(
                className,
                handleBasicClasses({
                    prefix: CLASSNAME,
                    orientation: orientation ?? (wrap || hAlign || vAlign ? Orientation.horizontal : null),
                    vAlign,
                    hAlign,
                    gap,
                }),
                wrap && `${CLASSNAME}--wrap`,
                fillSpace && `${CLASSNAME}--fill-space`,
                noShrink && `${CLASSNAME}--no-shrink`,
                marginAuto && castArray(marginAuto).map((align) => `${CLASSNAME}--margin-auto-${align}`),
            )}
        >
            {children}
        </div>
    );
});
FlexBox.displayName = COMPONENT_NAME;
FlexBox.className = CLASSNAME;
