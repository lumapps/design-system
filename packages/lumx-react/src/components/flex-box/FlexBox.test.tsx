import { mount, shallow } from 'enzyme';
import 'jest-enzyme';
import React, { ReactElement } from 'react';
import { commonTestsSuite, itShouldRenderStories, Wrapper } from '@lumx/react/testing/utils';

import { FlexBox, FlexBoxProps } from './FlexBox';
import * as stories from './FlexBox.stories';

const CLASSNAME = FlexBox.className as string;

type SetupProps = Partial<FlexBoxProps>;

/**
 * Mounts the component and returns common DOM elements / data needed in multiple tests further down.
 */
const setup = (propsOverride: SetupProps = {}, shallowRendering = true) => {
    const props: FlexBoxProps = {
        children: null,
        ...propsOverride,
    };
    const renderer: (el: ReactElement) => Wrapper = shallowRendering ? shallow : mount;
    const wrapper = renderer(<FlexBox {...props} />);

    return { props, wrapper };
};

describe(`<${FlexBox.displayName}>`, () => {
    // 1. Test render via snapshot.
    describe('Snapshots and structure', () => {
        itShouldRenderStories(stories, FlexBox);
    });

    /////////////////////////////

    // 2. Test defaultProps value and important props custom values.
    describe('Props', () => {
        // Nothing to do here.
    });

    /////////////////////////////

    // 3. Test events.
    describe('Events', () => {
        // Nothing to do here.
    });

    /////////////////////////////

    // 4. Test conditions (i.e. things that display or not in the UI based on props).
    describe('Conditions', () => {
        // Nothing to do here.
    });

    /////////////////////////////

    // 5. Test state.
    describe('State', () => {
        // Nothing to do here.
    });

    /////////////////////////////

    // Common tests suite.
    commonTestsSuite(setup, { className: 'wrapper', prop: 'wrapper' }, { className: CLASSNAME });
});
