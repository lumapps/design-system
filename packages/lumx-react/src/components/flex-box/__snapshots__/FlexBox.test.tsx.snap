// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlexBox> Snapshots and structure should render story 'Align' 1`] = `
Array [
  <div
    className="lumx-flex-box lumx-flex-box--orientation-horizontal lumx-flex-box--v-align-center lumx-flex-box--h-align-center"
  >
    <Button
      emphasis="high"
      size="m"
      style={
        Object {
          "height": 200,
        }
      }
      theme="light"
    >
      Button
    </Button>
    <FlexBox
      style={
        Object {
          "height": "fit-content",
        }
      }
    >
      Some text
    </FlexBox>
  </div>,
  <div
    className="lumx-flex-box"
    style={
      Object {
        "height": "fit-content",
      }
    }
  >
    Some text
  </div>,
]
`;

exports[`<FlexBox> Snapshots and structure should render story 'Flex' 1`] = `
Array [
  <div
    className="lumx-flex-box"
  >
    <FlexBox>
      <Icon
        icon="M20.71,7.04C21.1,6.65 21.1,6 20.71,5.63L18.37,3.29C18,2.9 17.35,2.9 16.96,3.29L15.12,5.12L18.87,8.87M3,17.25V21H6.75L17.81,9.93L14.06,6.18L3,17.25Z"
      />
    </FlexBox>
    <FlexBox>
      Some text in a div
    </FlexBox>
    <FlexBox>
      <Button
        emphasis="high"
        size="m"
        theme="light"
      >
        OK
      </Button>
    </FlexBox>
  </div>,
  <div
    className="lumx-flex-box"
  >
    <Icon
      icon="M20.71,7.04C21.1,6.65 21.1,6 20.71,5.63L18.37,3.29C18,2.9 17.35,2.9 16.96,3.29L15.12,5.12L18.87,8.87M3,17.25V21H6.75L17.81,9.93L14.06,6.18L3,17.25Z"
    />
  </div>,
  <div
    className="lumx-flex-box"
  >
    Some text in a div
  </div>,
  <div
    className="lumx-flex-box"
  >
    <Button
      emphasis="high"
      size="m"
      theme="light"
    >
      OK
    </Button>
  </div>,
]
`;

exports[`<FlexBox> Snapshots and structure should render story 'HorizontalFlex' 1`] = `
Array [
  <div
    className="lumx-flex-box lumx-flex-box--orientation-horizontal"
    style={
      Object {
        "border": "1px solid red",
        "height": "300px",
      }
    }
  >
    <Button
      emphasis="high"
      size="m"
      theme="light"
    >
      Default
    </Button>
    <FlexBox
      hAlign="top"
    >
      <Button
        emphasis="high"
        size="m"
        theme="light"
      >
        Top
      </Button>
    </FlexBox>
    <FlexBox
      fillSpace={true}
      hAlign="center"
      vAlign="center"
    >
      <Button
        emphasis="high"
        size="m"
        theme="light"
      >
        Center button
      </Button>
    </FlexBox>
    <FlexBox
      hAlign="bottom"
    >
      <Button
        emphasis="high"
        size="m"
        theme="light"
      >
        Bottom
      </Button>
    </FlexBox>
  </div>,
  <div
    className="lumx-flex-box lumx-flex-box--orientation-horizontal lumx-flex-box--h-align-top"
  >
    <Button
      emphasis="high"
      size="m"
      theme="light"
    >
      Top
    </Button>
  </div>,
  <div
    className="lumx-flex-box lumx-flex-box--orientation-horizontal lumx-flex-box--v-align-center lumx-flex-box--h-align-center lumx-flex-box--fill-space"
  >
    <Button
      emphasis="high"
      size="m"
      theme="light"
    >
      Center button
    </Button>
  </div>,
  <div
    className="lumx-flex-box lumx-flex-box--orientation-horizontal lumx-flex-box--h-align-bottom"
  >
    <Button
      emphasis="high"
      size="m"
      theme="light"
    >
      Bottom
    </Button>
  </div>,
]
`;

exports[`<FlexBox> Snapshots and structure should render story 'NoShrinkFlex' 1`] = `
Array [
  <div
    className="lumx-flex-box lumx-flex-box--orientation-horizontal"
    style={
      Object {
        "border": "1px solid red",
        "width": "150px",
      }
    }
  >
    <Button
      emphasis="high"
      size="m"
      theme="light"
    >
      Button
    </Button>
    <FlexBox
      noShrink={true}
    >
      Some long text
    </FlexBox>
    <Button
      emphasis="high"
      size="m"
      theme="light"
    >
      Button
    </Button>
  </div>,
  <div
    className="lumx-flex-box lumx-flex-box--no-shrink"
  >
    Some long text
  </div>,
]
`;

exports[`<FlexBox> Snapshots and structure should render story 'VerticalFlex' 1`] = `
Array [
  <div
    className="lumx-flex-box lumx-flex-box--orientation-vertical"
    style={
      Object {
        "border": "1px solid red",
        "height": "300px",
      }
    }
  >
    <Button
      emphasis="high"
      size="m"
      theme="light"
    >
      Default
    </Button>
    <FlexBox
      vAlign="left"
    >
      <Button
        emphasis="high"
        size="m"
        theme="light"
      >
        Left
      </Button>
    </FlexBox>
    <FlexBox
      fillSpace={true}
      hAlign="center"
      vAlign="center"
    >
      <Button
        emphasis="high"
        size="m"
        theme="light"
      >
        Center button
      </Button>
    </FlexBox>
    <FlexBox
      vAlign="right"
    >
      <Button
        emphasis="high"
        size="m"
        theme="light"
      >
        Right
      </Button>
    </FlexBox>
  </div>,
  <div
    className="lumx-flex-box lumx-flex-box--orientation-horizontal lumx-flex-box--v-align-left"
  >
    <Button
      emphasis="high"
      size="m"
      theme="light"
    >
      Left
    </Button>
  </div>,
  <div
    className="lumx-flex-box lumx-flex-box--orientation-horizontal lumx-flex-box--v-align-center lumx-flex-box--h-align-center lumx-flex-box--fill-space"
  >
    <Button
      emphasis="high"
      size="m"
      theme="light"
    >
      Center button
    </Button>
  </div>,
  <div
    className="lumx-flex-box lumx-flex-box--orientation-horizontal lumx-flex-box--v-align-right"
  >
    <Button
      emphasis="high"
      size="m"
      theme="light"
    >
      Right
    </Button>
  </div>,
]
`;

exports[`<FlexBox> Snapshots and structure should render story 'WrapFlex' 1`] = `
Array [
  <div
    className="lumx-flex-box lumx-flex-box--orientation-horizontal lumx-flex-box--wrap"
    style={
      Object {
        "border": "1px solid red",
        "width": "150px",
      }
    }
  >
    <FlexBox
      fillSpace={true}
      orientation="vertical"
    >
      <Button
        emphasis="high"
        size="m"
        theme="light"
      >
        Button
      </Button>
    </FlexBox>
    <Button
      emphasis="high"
      size="m"
      theme="light"
    >
      Button
    </Button>
    <Button
      emphasis="high"
      size="m"
      theme="light"
    >
      Button
    </Button>
  </div>,
  <div
    className="lumx-flex-box lumx-flex-box--orientation-vertical lumx-flex-box--fill-space"
  >
    <Button
      emphasis="high"
      size="m"
      theme="light"
    >
      Button
    </Button>
  </div>,
]
`;
